operator/main.go

```diff
@@ -84,7 +85,9 @@ func run() error {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+	config := ctrl.GetConfigOrDie()
+	config.UserAgent = defaults.UserAgent()
```

Comment:
The client has function `func AddUserAgent(config *Config, userAgent string) *Config {` maybe we should use that one instead?

Reply:
Implementation of AddUserAgent:
```go
func AddUserAgent(config *Config, userAgent string) *Config {
	fullUserAgent := DefaultKubernetesUserAgent() + "/" + userAgent
	config.UserAgent = fullUserAgent
	return config
}
```
Looks to me like this would only append the new string to the initial malformed string and not replace it.

Reply:
Well, this is the interface provided by the library, so it should be the default way of using it. If the default way is broken, then maybe we need to check newer versions or submit a patch?

Reply:
Just a small addition: by no means this is blocking this PR, I just noticed a potential to use the official API and wanted to make a note :) It is fine for me to move on with your approach if you feel that it is the right one :)

Reply:
I would not say the default way is broken, but in order to set it correctly one would have to set variables in the client-go module using -ldflags for our go build. After discussion with the team we decided to set in manually because that is more explicit and thus easier to understand. Otherwise it would be somewhere hidden in the build process and if someone sees it out of context it's hard to guess why we did that.